docs/operators/s3outputmodule.md

```diff
@@ -0,0 +1,199 @@
+
+-   ***mergerCount***   -   int
+    -   Specify the number of instances of S3FileMerger operator.
+    -   Default value = 1
+    
+-   ***timeOutWIndowCount***    -   int
+    -   Specify the count as streaming windows. This value will set to the operators in this module because the operators in this module is mostly interacts with the Amazon S3. Due to this reason, window id of these operators might be lag behind with the upstream operators.
+    -   Default value = 6000
+
+### Ports
+---------
+
+-   ***filesMetadataInput***    -   AbstractFileSplitter.FileMetadata
+    -   Input port for files metadata.
+    -   Mandatory
+
+-   ***blocksMetadataInput***   -   BlockMetadata.FileBlockMetadata
+    -   Input port for blocks metadata.
+    -   Mandatory
+    
+-   ***blockData*** -   AbstractBlockReader.ReaderRecord<Slice>
+    -   Input port for blocks data.
+    -   Mandatory
+    
+### Application Example
+-----------------------
+
+Please refer to [Example](https://github.com/DataTorrent/examples/tree/master/tutorials/s3output) for S3OutputModule sample application.
+
+### Partitioning
+----------------
+
+Partitioning the module means that the operators in the module can be partitioned.
+
+#### Stateless Partitioning
+---------------------------
+
+Partitioning the operator in module can be achieved as follows:
+
+-   ***S3InitiateFileUploadOperator***
+    -   Partition of this operator is achieved indirectly as follows:
+```xml
+<property>
+  <name>dt.operator.{ModuleName}#InitiateUpload.attr.PARTITIONER</name>
+  <value>com.datatorrent.common.partitioner.StatelessPartitioner:{N}</value>
+</property>     
+```
+where {ModuleName} is the name of the S3OutputModule and
+      {N} is the number of static partitions.
+Above lines will partition S3InitiateFileUploadOperator statically {N} times.
+
+-   ***S3BlockUploadOperator***
+    -   Locality of S3BlockUploadOperator with upstream operator (FSInputModule/BlockReader) must set to PARTITION_PARALLEL for performance benefits by avoiding serialization/deserialization of objects. So, partitioning of this operator depends on upstream operator which is of type FSInputModule/BlockReader. 
+    
+-   ***S3FileMerger***
+    -   By setting the parameter "mergerCount", **S3FileMerger** be statically partitioned. This can be achieved by two ways:
+
+    -   (a) Following code can be added to populateDAG(DAG dag, Configuration conf) method of application to statically partitioning **S3FileMerger** {N} times:
+    ```java
+      FSInputModule inputModule = dag.addModule("HDFSInputModule", new FSInputModule());
+      S3OutputModule outputModule = dag.addModule("S3OutputModule", new S3OutputModule());
+      outputModule.setMergerCount({N});
+    ```
+    
+    -   (b) By setting the parameter in properties file as follows
+    ```xml
+    <property>
+      <name>dt.operator.{ModuleName}.prop.mergerCount</name>
+      <value>{N}</value>
+    </property>
+    ```
```

Comment:
Please review this document using mkdocs and adjust the spacing and markdown so it renders properly. The above Java and XML do not show correctly.